Xplat/src/main/java/vazkii/botania/common/block/flower/functional/LooniumBlockEntity.java

```diff
@@ -78,116 +92,119 @@ public void tickFlower() {
 			return;
 		}
 
-		if (ticksExisted == 1 && lootTables == DEFAULT_LOOT_TABLES) {
-			autodetectStructureLootTables(world);
+		if (detectedStructures == null) {
+			detectStructure(world);
 		}
 
-		if (redstoneSignal == 0 && ticksExisted % 100 == 0
-				&& getMana() >= COST && world.getDifficulty() != Difficulty.PEACEFUL) {
-			var rand = world.random;
+		if (redstoneSignal != 0 || ticksExisted % 100 != 0 || world.getDifficulty() == Difficulty.PEACEFUL) {
+			return;
+		}
 
-			ItemStack stack = pickRandomLoot(world, rand);
-			if (stack.isEmpty()) {
-				return;
-			}
+		var configData = BotaniaAPI.instance().getConfigData();
+		var structureConfigs = determineStructureConfigs(configData, detectedStructures);
+		var lootTables = determineLootTables(world, detectedStructures.keySet());
 
-			int bound = RANGE * 2 + 1;
-			int xp = getEffectivePos().getX() - RANGE + rand.nextInt(bound);
-			int yp = getEffectivePos().getY();
-			int zp = getEffectivePos().getZ() - RANGE + rand.nextInt(bound);
+		if (lootTables.isEmpty()) {
+			return;
+		}
 
-			BlockPos pos = new BlockPos(xp, yp - 1, zp);
-			do {
-				pos = pos.above();
-				if (pos.getY() >= world.getMaxBuildHeight()) {
-					return;
-				}
-			} while (world.getBlockState(pos).isSuffocating(world, pos));
-			pos = pos.above();
+		var randomPick = lootTables.get(world.random.nextInt(lootTables.size()));
+		var pickedConfig = structureConfigs.getOrDefault(randomPick.key(),
+				structureConfigs.get(LooniumStructureConfiguration.DEFAULT_CONFIG_ID));
+		var pickedLootTable = randomPick.value();
 
-			double x = pos.getX() + Math.random();
-			double y = pos.getY() + Math.random();
-			double z = pos.getZ() + Math.random();
+		if (getMana() < pickedConfig.manaCost) {
```

Comment:
Maybe filter to only pick configs we can afford? Or maybe we explicitly don't want this, because i guess it's kinda cheesable? Idk, open to discussion

Reply:
I think it's best to just pick one, and then do nothing if we can't afford it, even if there are other configs we can afford. So keep current behavior as it's implemented here